src/ws_server.rs

```diff
@@ -30,8 +34,40 @@ impl WsServer {
 
 impl<T: Controller> Handler for WsHandler<T> {
 
-    fn on_open(&mut self, _: Handshake) -> Result<()> {
+    fn on_open(&mut self, handshake: Handshake) -> Result<()> {
         info!("Hello new ws connection");
+
+        let resource = &handshake.request.resource()[..];
+        let url = Url::parse(&format!("http://fox.box{}", resource)).unwrap();
```

Comment:
Don't use `unwrap` here, please. If parsing fails close the connection and bail out.